### DISCUSSION TITLE
Is there a way to create a hasOne with the foreign key on the parent model?

### DISCUSSION THREAD

**Discussion by nelsonjd** [adonisjs/core#4258] [Help]
Is this possible?
  ```
  Model A
  @column()
  public cashWeightingId: number

  @hasOne(() => CashWeighting)
  public cashWeighting: HasOne<typeof CashWeighting>
```
Or are you forced to call this a belongsTo in which case it must be the "child" model?

Right now with this as a hasOne, it is complaining that it is expecting `modelAId` to exist on CashWeighting.
 
I know I could call this a belongsTo and things will work fine.  Although a different ORM api is available for belongs_to vs. has_one, and the mental model of "who is the parent" is different, but that is just a mental abstraction.

I thought this hasOne setup was possible because of this answer given by Harminder: https://github.com/adonisjs/core/discussions/1264

**Comment by LuizCristino** [comment]
You can try something like this, although I don't recommend for readability and understanding of your DER

```ts
class ModelA extends BaseModel {
  @column({ isPrimary: true })
  public id: number;

  @column({ serializeAs: 'model_b_id' })
  public modelBId: number
  
  @hasOne(() => ModelA, { localKey: 'model_b_id', foreignKey: 'id' })
  public modelB: HasOne<typeof ModelB>;
}

class ModelB extends BaseModel {
  @column({ isPrimary: true })
  public id: number;
  
  @hasMany(() => ModelA, { localKey: 'id', foreignKey: 'model_b_id' })
  public modelsA: HasMany<typeof ModelA>;
}
